generator/models.py

```diff
@@ -200,17 +201,67 @@ def save(self, *args, **kwargs):
         super(Section, self).save(*args, **kwargs)
 
 
+class Papersize(models.Model):
+    """
+    This class holds paper information for generating a songbook.
+    All size are in millimetters..
+    """
+
+    name = models.CharField(max_length=200,
+                            verbose_name=_(u"nom du format")
+                            )
+
+    width = models.IntegerField()
+    height = models.IntegerField()
+
+    left = models.IntegerField()
+    right = models.IntegerField()
+    top = models.IntegerField()
+    bottom = models.IntegerField()
+    bindingoffset = models.IntegerField()
+
+    class Meta:
+        verbose_name = _("Papier")
+
+    def clean(self):
+        # Don't allow draft entries to have a pub_date.
+        if self.width > self.height:
+            raise ValidationError(_('Les dimensions doivent être donné pour un format portrait (hauteur >= largeur).'))
+    def __str__(self):
+        return self.name
+
+    def get_geometry(self):
+        """Return a list containing the geometry properties for the papersize"""
+        
+        geometry = []
+
+        geometry.append("paperwidth=" + str(self.width) + "mm")
+        geometry.append("paperheight=" + str(self.height) + "mm")
+        
+        geometry.append("asymmetric")
+        
+        fields = [
+            'top',
+            'left',
+            'bottom',
+            'right',
+            'bindingoffset',
+        ]
+        for field in fields:
+            geometry.append(field + "=" + str(getattr(self, field)) + "mm")
```

Comment:
Je pense qu'on peut mettre les valeurs que l'on veut (cm, in, mm...).
Mais dans une BDD, je préfère stocker des entiers (et arbitrairement, j'ai choisi les mms...)
Que veux-tu dire avec:
> attribut des champs de ce modèle